quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/MappedTripleGroupServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+        }
+        return mappedTripleGroupRepository.findById(mappedTripleGroupObjectId).orElse(null);
+    }
+
+    /**
+     * This method looks for a MappedTripleGroup using the given mappedTripleGroupId
+     * and updates the name of that MappedTripleGroup with the given name.
+     * 
+     * @param collectionId used to look for the MappedTripleGroup entry
+     * @return the updated MappedTripleGroup entry
+     * @throws InvalidObjectIdException          if collectionId couldn't be
+     *                                           converted to ObjectId
+     * @throws MappedTripleGroupNotFoundException if MappedTripleGroup by the given
+     *                                           collectionId was not found
+     * @throws CollectionNotFoundException       if collection with given
+     *                                           collectionId doesn't exist
+     */
+    @Override
+    public MappedTripleGroup updateName(String mappedTripleGroupId, String name)
+            throws InvalidObjectIdException, MappedTripleGroupNotFoundException {
+        MappedTripleGroup mappedTripleGroup = getById(mappedTripleGroupId);
+        if (mappedTripleGroup != null) {
+            mappedTripleGroup.setName(name);
+            return mappedTripleGroupRepository.save(mappedTripleGroup);
+        } else {
+            throw new MappedTripleGroupNotFoundException("MappedTripleGroupId: " + mappedTripleGroupId);
+        }
+    }
+    
+    /**
+     * This method will try to find a MappedTripleGroup entry based on the given
+     * collectionId and mappedTripleType. If no entry was found, it will try to create a new one.
+     * 
+     * @param collectionId used to look for the MappedTripleGroup entry
+     * @param mappedTripleType is the group type which is to be looked for in the database
+     * @return an existing or a new mappedTripleGroup entry
+     * @throws InvalidObjectIdException    if collectionId couldn't be converted to
+     *                                     ObjectId
+     * @throws CollectionNotFoundException if a collection for given collectionId
+     *                                     doesn't exist
+     */
+    @Override
+    public MappedTripleGroup get(String collectionId, MappedTripleType mappedTripleType)
+            throws InvalidObjectIdException, CollectionNotFoundException {
+        MappedTripleGroup mappedTripleGroup = findByCollectionIdAndMappingType(collectionId, mappedTripleType);
+
+        // In case this is a new collection, or existing collection but new mapping type for that collection
+        // we create a new MappedTripleGroup entry
+        if (mappedTripleGroup == null) {
+            mappedTripleGroup = add(collectionId, mappedTripleType);
+        }
+        if (mappedTripleGroup == null) {
+            logger.error("Couldn't find or persist a new MappedTripleGroup entry for collectionId: " + collectionId);
+        }
```

Comment:
this if is odd here. It just adds a log message but proceeds. That error message should be logged where the method is called an its result is handled.

Reply:
In fact, when does this situation occur?

Reply:
This situation could occur if something fails while inserting a new entry for mappedTripleGroup
But I can remove this altogether and just add the null check in the network submission request handler
Should I do that?

Reply:
I would expect that there would be an exception in that case?

Reply:
There's no exception raised by SimpleMongoRepository if saving fails
So, I have just added null check in the `AddNetworkApiController` itself and added logging there